core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/validator/EntityValidator.java

```diff
@@ -31,6 +35,14 @@
   public void validate(Entity<?> entity, Optional<? extends Entity<?>> existing) {
```

Comment:
I think the Optional here obscures the fact that there are two possible validation paths. I also don't think we should be using except as a parameter type.